core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java

```diff
@@ -358,57 +358,43 @@ public long handleWayTags(ReaderWay way, long allowed, long relationFlags) {
     }
 
     private long extractMeter(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
-        String value = extractRoadAttribute(way, keys);
-
+        String value = way.getFirstPriorityTag(keys);
         if (Helper.isEmpty(value)) return flags;
 
         double val;
         try {
             val = stringToMeter(value);
-        } catch (Throwable t) {
-            LOG.warn("Unable to extract meter from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), t);
-            val = 0.0;
+        } catch (Exception ex) {
+            LOG.warn("Unable to extract meter from malformed road attribute '{}' for way (OSM_ID = {}).", value, way.getId(), ex);
+            return flags;
         }
 
         try {
-            flags = valueEncoder.setDoubleValue(flags, val);
-        }
-        catch (IllegalArgumentException e) {
-            LOG.warn("Unable to process value '{}' for way (OSM_ID = {}).", val, way.getId(), e);
+            return valueEncoder.setDoubleValue(flags, val);
+        } catch (IllegalArgumentException e) {
+            LOG.warn("Unable to process meter value '{}' for way (OSM_ID = {}).", val, way.getId(), e);
+            return flags;
         }
-
-        return flags;
     }
 
     private long extractTons(ReaderWay way, long flags, EncodedDoubleValue valueEncoder, List<String> keys) {
-        String value = extractRoadAttribute(way, keys);
-
+        String value = way.getFirstPriorityTag(keys);
```

Comment:
This should be the same functionality (do we have a test covering a change like this?)